utils/io/fileutils/files_test.go

```diff
@@ -1,18 +1,74 @@
 package fileutils
 
 import (
-	biutils "github.com/jfrog/build-info-go/utils"
-	"github.com/jfrog/jfrog-client-go/utils/io"
 	"os"
 	"path/filepath"
 	"reflect"
 	"regexp"
 	"strings"
 	"testing"
 
+	biutils "github.com/jfrog/build-info-go/utils"
+	"github.com/jfrog/jfrog-client-go/utils/io"
+
 	"github.com/stretchr/testify/assert"
 )
 
+func TestIsPathExistsAndIsPathAccessible(t *testing.T) {
```

Comment:
for isAccessible should we create a file without permissions and test that it returns false as expected?

Reply:
I tried adding a test that creates a file without permissions, but you have no idea how challenging it is to test this. The major challenge comes from the fact that Windows and Linux/MacOS treat files without permissions completely differently. It is hard to make all OS work inside a single test. The only option I found was to split the test into two files, so that only one of them compilers for Windows. This is due to a package that can handle Windows locking. I ended up giving up and just testing the IsPathAccessible API with non existing files. It's just not worth the complexity @sverdlov93.